raster/r.colors.out/local_proto.h

```diff
@@ -0,0 +1,11 @@
+#include <grass/gis.h>
+#include <grass/parson.h>
+
+enum ColorFormat { RGB, HEX, HSV, DEFAULT };
```

Comment:
May I suggest to change the name of the constant `DEFAULT` to `XTERM`, which I believe is a better representation of the format. (See e.g. https://news.ycombinator.com/item?id=35138390 for some introductory discussion). `XTERM` may then all the same be the "default" format.

Reply:
@nilason I can't find any examples of RRR:GGG:BBB being XTERM. The definition in the ctlseqs document seems more complex than that. I was not able to find out what "ISO-8613-6 direct color" means (referenced by ctlseqs).

Reply:
XTERM primarily uses the semicolon format, which is more widely supported and recognized (38;2;R;G;B for RGB colors). According to XTERM's documentation (ctlseqs.txt), XTERM does allow colons in place of semicolons to align with certain interpretations of the ISO 8613 standard, though this is not widely adopted. The ctlseqs.txt notes that “ISO-8613-3 can be interpreted in more than one way,” allowing either semicolons or colons, but there is no standard agreement on using RRR:GGG specifically for XTERM.

Reply:
So, what do you think? Should I use XTERM, default, or another format? Please suggest.
cc: @wenzeslaus @nilason @cwhite911

Reply:
How about using `crgb` instead of xterm, which is short for `colon-separated` RGB (RRR:GGG:BBB)? Although this is just a made-up name, I searched the web and couldn't find a standard format name for RRR:GGG:BBB.

Reply:
Thank you for the details. Well, the case for xterm is weak. I'm against `default` because I think we should not use the word default as value and that a default value here should be `hex`. I think in the past, we called it GRASS color triplet. `triplet`?

Reply:
Yeah, I think "triplet" would be fine, since I can see many instances in the codebase where we refer to R:G:B as "triplet."

Reply:
My immediate goal with my suggestion to replace "DEFAULT" with "XTERM", was to avoid the use of "DEFAULT" as an option in itself (and not stand-in for a "proper" format). Furthermore, at that time it was only to intended to be used in-code.

I'm still quite positive the format `r:g:b` has its origin in XTERM, although it is difficult to find a quote on that. Internally, GRASS used space-separated rgb until version 4, which with its release in 1991 introduced the colon-separated.

Compare:

https://github.com/OSGeo/grass-legacy/blob/grass_3_2_0/src/libes/gis/colors.c

https://github.com/OSGeo/grass-legacy/blob/grass_4_0_0/src/libes/gis/color_write.c

`r.colors`, lagging behind, was updated with version 5.5 (in 2004) for consistency ([commit 9102](https://trac.osgeo.org/grass/browser/grass/branches/releasebranch_5_5/src/raster/r.colors/cmd/rules.c?rev=9102)).

I don't insist on `xterm` as it is not absolutely clear whether it uses colon or semi-colon. But I don't like `triplet` either, as it equally doesn't tell us anything regarding format. A triplet may just as well be space separated; [Emacs man](https://www.gnu.org/software/emacs/manual/html_node/emacs/Colors.html) reference the hex format of '#RRGGBB' as _"RGB triplet"_.

May I suggest something more descriptive like `col_sep_rgb`, `colsep_rgb`, or `colon_sep_rgb`?!

Reply:
(In GRASS docs the triplet is referenced as "R:G:B triplet", which does make sense, but that is not the same as "triplet", or "RGB triplet".)

Reply:
@wenzeslaus @cwhite911, apologies for the uncertainty—just wanted to confirm what you suggest I should use, considering the conversation above.

Reply:
Need some confirmation on what needs to be done here...
@petrasovaa